torch/fx/experimental/symbolic_shapes.py

```diff
@@ -323,18 +323,32 @@ def constrain_range(a, *, min: Optional[int], max: Optional[int] = None):
     if not isinstance(a, SymInt):
         if not (min <= a <= max):
             raise ValueRangeError(f"Invalid value {a} for range [{min}:{max}]")
-        return
-    if isinstance(a.node.expr, sympy.Integer):
+    elif isinstance(a.node.expr, sympy.Integer):
         if not (min <= int(a.node.expr) <= max):
             raise ValueRangeError(f"Invalid value {int(a.node.expr)} for range [{min}:{max}]")
-        return
-    assert isinstance(a.node.expr, sympy.Symbol), "constraining non-Symbols NYI"
+    elif isinstance(a.node.expr, sympy.Symbol):
+        pass
+    else:
+        raise ValueRangeError("constraining non-Symbols NYI")
+
     # TODO: Shouldn't we install a guard if the symbol is backed?  Or is the
     # semantics that this is an "unchecked" assert (but it this actually
     # something useful?  Might be better to restrict only for unbacked
     # SymInt).
-    r = a.node.shape_env.var_to_range[a.node.expr]
-    a.node.shape_env.var_to_range[a.node.expr] = ValueRanges(
+    if not isinstance(a, SymInt):
```

Comment:
Why not pull this into the body of L323? Then the remaining edits above (i.e., adding `elif` / `else` to avoid early `return`) will not be necessary as well.

Reply:
Yup you're right